### PR TITLE
Wrap cipher in code blocks

### PR DESCRIPTION
**Pull request by sean-yeoh**
### Summary
This PR wraps ciphers inside issues so that it doesn't lead to weird export errors.

### Copyright assignment

> I assign all rights, including copyright, to any future Dradis
> work by myself to Security Roots.

### CODE REVIEW

File: lib/dradis/plugins/qualys/importer.rb [hunk 1]
```diff
@@ -1,5 +1,6 @@
 module Dradis::Plugins::Qualys
   class Importer < Dradis::Plugins::Upload::Importer
+    SSL_CIPHER_VULN_IDS = %w[38140 38141 42366].freeze
```
Comment (aapomm) [inline lib/dradis/plugins/qualys/importer.rb]: Can you add a comment to point towards a reference to what this numbers could mean?

File: lib/dradis/plugins/qualys/importer.rb [hunk 2]
```diff
@@ -78,11 +79,20 @@ def process_vuln(collection, vuln_number, xml_cat)
...
+
       issue = content_service.create_issue(text: issue_text, id: vuln_number)
 
       logger.info{ "\t\t => Creating new evidence" }
       evidence_content = template_service.process_template(template: 'evidence', data: xml_cat)
       content_service.create_evidence(issue: issue, node: self.host_node, content: evidence_content)
     end
+
+    def add_bc_to_ssl_cipher_list(text)
+      text.gsub(/^(.*?):!(.*?)$/) { "\nbc. #{$1}:!#{$2}\n" }
```
Comment (aapomm) [inline lib/dradis/plugins/qualys/importer.rb]: Are we sure that this syntax will work for all cipher list? I'm under the impression that the ciphers are delimited by `:` and the `!` here is used for special cases like: `!EXPORT` `!LOW` `!aNULL` `!eNULL` `!SSLv2` which I assume wouldn't exist in all cipher list.

File: lib/dradis/plugins/qualys/importer.rb (hunk 1 above)
Comment (sean-yeoh) [reply lib/dradis/plugins/qualys/importer.rb]: Will do

File: lib/dradis/plugins/qualys/importer.rb (hunk 2 above)
Comment (sean-yeoh) [reply lib/dradis/plugins/qualys/importer.rb]: > Are we sure that this syntax will work for all cipher list? 

I don't think so. The regex will be quite complicated if we want to capture all ciphers.

File: spec/qualys/asset/importer_spec.rb [hunk 3]
```diff
@@ -20,7 +20,7 @@ module Dradis::Plugins
     let(:run_import!) { @importer.import(file: example_xml) }
 
     it 'creates nodes as needed' do
-      expect_to_create_node_with(label: '10.0.2.8')
+      expect_to_create_node_with(label: '10.0.0.1')
```
Comment (sean-yeoh) [inline spec/qualys/asset/importer_spec.rb:23]: Not related but this was a failing spec.

**Review by etdsoft — COMMENTED** [review]
If this works I'm happy to merge, the L96-L109 code in `element.rb` is getting a bit unwieldy, and we could consider a refactor (if simple enough), I'd merge and try the refactor in a separate PR:

From:
```ruby
vuln_id = @xml.attributes['number'].to_s
tag = @xml.at_xpath("./#{method_name.upcase}")
if tag && !tag.text.blank?
  if tags_with_html_content.include?(method)
    result = Qualys::cleanup_html(tag.text)
    result = add_bc_to_ssl_cipher_list(result) if SSL_CIPHER_VULN_IDS.include?(vuln_id)
    return result
  else
    return tag.text
  end
else
  'n/a'
end
```

To something like:
```ruby
tag = @xml.at_xpath("./#{method_name.upcase}")
if method_name == 'qualys_collection'
  @xml.name
elsif tag && !tag.text.blank?
  vuln_id = @xml.attributes['number'].to_s
  cleanup_tag(method, vuln_id, tag.text)
else
  # nothing found, the tag is valid but not present in this ReportItem
  return nil
end


private
def cleanup_tag(method, vuln_id, text)
  result = text
  result = Qualys::cleanup_html(result) if tags_with_html_content.include?(method)
  result = add_bc_to_ssl_cipher_list(result) if SSL_CIPHER_VULN_IDS.include?(vuln_id)
  result
end
```

**Pull request merged**